testproject/Assets/ManualTests/Scripts/ConnectionApprovalComponent.cs

```diff
@@ -0,0 +1,130 @@
+using System.Collections;
+using System.Text;
+using UnityEngine;
+using UnityEngine.UI;
+using MLAPI;
+
+public class ConnectionApprovalComponent : NetworkBehaviour
+{
+    [SerializeField]
+    private string m_ApprovalToken;
+
+    [SerializeField]
+    private uint m_GlobalObjectIdHashOverride;
+
+    [SerializeField]
+    private Text m_ConnectionMessageToDisplay;
+
+    [SerializeField]
+    private Toggle m_SimulateFailure;
+
+    [SerializeField]
+    private Toggle m_PlayerPrefabOverride;
+
+
+
+    private void Start()
+    {
+        
+        if (m_PlayerPrefabOverride)
+        {
+            m_PlayerPrefabOverride.gameObject.SetActive(false);
+        }
+
+        if (m_SimulateFailure)
+        {
+            m_SimulateFailure.gameObject.SetActive(false);
+        }
+
+        if (m_ConnectionMessageToDisplay)
+        {
+            m_ConnectionMessageToDisplay.gameObject.SetActive(false);
+        }
+
+        if (NetworkManager != null && NetworkManager.NetworkConfig.ConnectionApproval)
+        {
+            NetworkManager.ConnectionApprovalCallback += ConnectionApprovalCallback;
+
+            if (m_ApprovalToken != string.Empty)
+            {
+                NetworkManager.NetworkConfig.ConnectionData = Encoding.ASCII.GetBytes(m_ApprovalToken);
+            }
+            else
+            {
+                Debug.LogError($"You need to set the {nameof(m_ApprovalToken)} to a value first!");
+            }
+
+            NetworkManager.OnClientDisconnectCallback += NetworkManager_OnClientDisconnectCallback;
+        }
+    }
+
+    public override void NetworkStart()
+    {
+        if (m_SimulateFailure)
+        {
+            m_SimulateFailure.gameObject.SetActive(IsServer);
+        }
+
+        if (m_PlayerPrefabOverride)
+        {
+            m_PlayerPrefabOverride.gameObject.SetActive(IsServer);
+        }
+    }
+
+    private void NetworkManager_OnClientDisconnectCallback(ulong obj)
+    {
+        Debug.Log($"Client {obj} connected!");
+    }
+
+    private void ConnectionApprovalCallback(byte[] arg1, ulong arg2, NetworkManager.ConnectionApprovedDelegate arg3)
```

Comment:
extremely minor: I think args should start from zero like `arg0, arg1, arg2` :D

Reply:
It was the default values.  :)